pyrtl/simulation.py

```diff
@@ -793,14 +793,48 @@ def add_fast_step(self, fastsim):
         for wire_name in self.trace:
             self.trace[wire_name].append(fastsim.context[wire_name])
 
-    def print_trace(self, file=sys.stdout):
+    def print_trace(self, file=sys.stdout, base=10, compact=False):
+        """
+        Prints a list of wires and their current values.
+        :param int base: the base the values are to be printed in
+        :param bool compact: whether to omit spaces in output lines
+        """
         if len(self.trace) == 0:
             raise PyrtlError('error, cannot print an empty trace')
-        maxlen = max(len(w) for w in self.trace)
-        for w in sorted(self.trace, key=_trace_sort_key):
-            file.write(' '.join([w.rjust(maxlen),
-                       ''.join(str(x) for x in self.trace[w])+'\n']))
-            file.flush()
+        if base not in (2, 8, 10, 16):
+            raise PyrtlError('please choose a valid base')
+
+        def dec(i):
+            return '0d' + str(i)
+
+        maxlenleft = max(len(w) for w in self.trace)
+        if compact:
+            for w in sorted(self.trace, key=_trace_sort_key):
+                file.write(' '.join([w.rjust(maxlenleft),
+                                     ''.join(str(x) for x in self.trace[w]) + '\n']))
+                file.flush()
+
+        else:
+            bfunc = dec
+            if base == 2:
+                bfunc = bin
+            elif base == 8:
+                bfunc = oct
+            elif base == 16:
+                bfunc = hex
+            maxlenval = max(max((len(bfunc(x)[(1 if bfunc == oct else 2):])
+                                 for x in self.trace[w])) for w in self.trace)
```

Comment:
The intersting thing is that you can actually remove the inside `max` and it will work (i think you'll also need to remove the both sets of parens (the ones in between `max` and `len`))